Modules/posixmodule.c

```diff
@@ -8588,6 +8588,49 @@ os_pidfd_open_impl(PyObject *module, pid_t pid, unsigned int flags)
 #endif
 
 
+#if defined(__linux__) && defined(HAVE_SETNS)
```

Comment:
There is no need to check for ``__linux__``. ``HAVE_SETNS`` is good enough. If you think it is not sufficient, then you should rather change the configure script and only check for the functions on Linux.

```suggestion
#ifdef HAVE_SETNS
```